portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -24,9 +24,34 @@
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
+  private ArrayList<String> messages = new ArrayList<>();
+  
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello world!</h1>");
+    // response.setContentType("text/html;");
+    // response.getWriter().println("Hello Erica!");
+
+    messages.add("Hello!");
+    messages.add("Hola!");
+    messages.add("Hallo!");
+
+    String json = convertToJson(messages);
+
+    response.setContentType("application/json;");
+    response.getWriter().println(json);
+  }
+
+  private String convertToJson(ArrayList messages) {
+      String json = "[";
+
+    //   for (String message: messages) {
+    //       json += "{\"comment\": " + message + "}";
+    //   }
+
+      json += "{\"message\": \"" + messages.get(0) +"\"}, ";
```

Comment:
json spec requires that certain characters, such as double-quotes and backslashes, in strings are escaped properly. It's probably better to use a json encoding library to create this output unless you're feeling up to the task of rolling your own!

Reply:
Could you elaborate on what you mean? From what there was in the walkthrough and the linked external help pages I thought I had formatted the json string correctly, but if it's incorrect or could cause problems in the future then I'd appreciate the help.

Reply:
Suppose somebody left a comment like
> this is a double quote -> "

Then this logic would take that and put another double quote after it to terminate the json string, but it was already terminated by the " from the comment, which wasn't escaped. A proper encoder would escape the "real" double quote by prefixing it with a \, so the json output would read like: `{"message":"this is a double quote -> \""}`

And note the escape character \ itself needs to be escaped if it's part of a message: `{"double quote": "\"", "backslash": "\\"}`

[Json is a relatively simple encoding scheme](https://www.json.org/) but it's definitely better to use a standard library. I'll LGTM for now since the messages are hard-coded to not contain anything needing escapes, but this will need to be addressed before allowing free-form comments to be posted+displayed